### PR TITLE
docs: mongodbindexer crud

### PR DESCRIPTION
**Pull request by florian-hoenicke**

### CODE REVIEW

File: indexers/keyvalue/MongoDBIndexer/__init__.py [hunk 1]
```diff
@@ -34,43 +37,68 @@ def post_init(self):
...
+        """Get the database handler.
+        """
         return self.handler
-    
+
     def add(self, keys: Iterator[int], values: Iterator[bytes], *args, **kwargs) -> None:
+        """Add JSON-friendly serialized documents to the indexer.
+
+        :param keys: document ids
+        :param values: JSON-friendly serialized documents
```
Comment (JoanFM) [inline indexers/keyvalue/MongoDBIndexer/__init__.py:60]: Then the type hint isnt a little missleading? are bytes a correct hint?

File: indexers/keyvalue/MongoDBIndexer/__init__.py [hunk 2]
```diff
@@ -34,43 +37,68 @@ def post_init(self):
...
     @cached_property
     def query_handler(self):
+        """Cached database handler.
+        """
         return self.get_query_handler()
 
     def query(self, key: int, *args, **kwargs) -> Optional[bytes]:
+        """Query the protobuf documents via id.
+        :param key: ``id``
+        :return: protobuf chunk or protobuf document
```
Comment (JoanFM) [inline indexers/keyvalue/MongoDBIndexer/__init__.py:81]: avoid usage of chunk

File: indexers/keyvalue/MongoDBIndexer/__init__.py [hunk 3]
```diff
@@ -9,11 +9,12 @@ class MongoDBIndexer(BinaryPbIndexer):
...
-    def __init__(self, 
-                 hostname: str = '127.0.0.1', 
-                 port: int = 27017, 
-                 username: str = None, 
-                 password: str = None, 
+    # TODO user name and password as environment variables
+    def __init__(self,
+                 hostname: str = '127.0.0.1',
+                 port: int = 27017,
+                 username: str = None,
```
Comment (JoanFM) [inline indexers/keyvalue/MongoDBIndexer/__init__.py:16]: can we add proper Optional type hint?

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: indexers/keyvalue/MongoDBIndexer/__init__.py (hunk 1 above)
Comment (florian-hoenicke) [reply indexers/keyvalue/MongoDBIndexer/__init__.py:60]: thanks I changed it to `serialized documents`

File: indexers/keyvalue/MongoDBIndexer/__init__.py (hunk 3 above)
Comment (florian-hoenicke) [reply indexers/keyvalue/MongoDBIndexer/__init__.py:16]: good idea

File: indexers/keyvalue/MongoDBIndexer/__init__.py (hunk 2 above)
Comment (florian-hoenicke) [reply indexers/keyvalue/MongoDBIndexer/__init__.py:81]: changed

**Comment by nan-wang** [comment]
The failure in the unit tests is due to the wrong setting of the entry point in the Dockerfile. Setting the entry point as below will leads to `_args` become invalid at `https://github.com/jina-ai/jina/blob/master/jina/peapods/runtimes/container/__init__.py#L129`

```
ENTRYPOINT [ "sh", "-c", "mongod --fork -logpath /var/log/mongod.log && jina pod --uses config.yml" ]
```

It should be changed to the following and anyhow we should **NOT** start the mongodb for the user @deepankarm 
```
ENTRYPOINT ["jina", "pod", "--uses", "config.yml"]
```

**Comment by nan-wang** [comment]
FYI, @florian-hoenicke  I am working on a commit to fix this.

**Comment by nan-wang** [comment]
`jina hub build`  does **NOT** like this pattern. We are revising the Dockerfile. https://github.com/jina-ai/jina/blob/master/jina/docker/hubio.py#L632

**Comment by nan-wang** [comment]
depends on https://github.com/jina-ai/jina/issues/1825

File: indexers/keyvalue/MongoDBIndexer/Dockerfile [hunk 4]
```diff
@@ -1,17 +1,20 @@
...
+RUN python3.7 -m pip install -r requirements.txt
+
+COPY . /workspace
+WORKDIR /workspace
 
 # for testing the image
-RUN mongod --fork -logpath /var/log/mongod.log && python3.8 -m pip install pytest && pytest -s
+RUN mongod --fork -logpath /var/log/mongod.log && python3.7 -m pip install pytest && pytest -s
 
 # starts mongod as a deamon & jina pod
```
Comment (cristianmtr) [inline indexers/keyvalue/MongoDBIndexer/Dockerfile:19]:
```suggestion
# starts jina pod
```

Comment (florian-hoenicke) [reply indexers/keyvalue/MongoDBIndexer/Dockerfile]: thanks, changed

**Review by JoanFM — APPROVED** [review]

**Review by cristianmtr — APPROVED** [review]

**Pull request merged**